src/scheduled_tasks.rs

```diff
@@ -59,24 +70,42 @@ pub fn setup(db_url: String, user_agent: String) -> Result<(), LemmyError> {
   }
 }
 
-/// Reindex the aggregates tables every one hour
-/// This is necessary because hot_rank is actually a mutable function:
-/// https://dba.stackexchange.com/questions/284052/how-to-create-an-index-based-on-a-time-based-function-in-postgres?noredirect=1#comment555727_284052
-fn reindex_aggregates_tables(conn: &mut PgConnection, concurrently: bool) {
-  for table_name in &[
-    "post_aggregates",
-    "comment_aggregates",
-    "community_aggregates",
-  ] {
-    reindex_table(conn, table_name, concurrently);
-  }
-}
-
-fn reindex_table(conn: &mut PgConnection, table_name: &str, concurrently: bool) {
-  let concurrently_str = if concurrently { "concurrently" } else { "" };
-  info!("Reindexing table {} {} ...", concurrently_str, table_name);
-  let query = format!("reindex table {concurrently_str} {table_name}");
-  sql_query(query).execute(conn).expect("reindex table");
+/// Update the hot_rank columns for the aggregates tables
+fn update_hot_ranks(conn: &mut PgConnection) {
+  info!("Updating hot ranks...");
+
+  // Only update for the last week of content
+  let last_week = now - diesel::dsl::IntervalDsl::weeks(1);
+
+  diesel::update(post_aggregates::table)
+    .filter(post_aggregates::published.gt(last_week))
+    .set((
+      post_aggregates::hot_rank.eq(hot_rank(post_aggregates::score, post_aggregates::published)),
+      post_aggregates::hot_rank_active.eq(hot_rank(
+        post_aggregates::score,
+        post_aggregates::newest_comment_time_necro,
+      )),
+    ))
+    .execute(conn)
+    .expect("update post_aggregate hot_ranks");
+
+  diesel::update(comment_aggregates::table)
+    .filter(comment_aggregates::published.gt(last_week))
+    .set(comment_aggregates::hot_rank.eq(hot_rank(
+      comment_aggregates::score,
+      comment_aggregates::published,
+    )))
+    .execute(conn)
+    .expect("update comment_aggregate hot_ranks");
+
+  diesel::update(community_aggregates::table)
+    .filter(community_aggregates::published.gt(last_week))
+    .set(community_aggregates::hot_rank.eq(hot_rank(
+      community_aggregates::subscribers,
+      community_aggregates::published,
+    )))
+    .execute(conn)
+    .expect("update community_aggregate hot_ranks");
   info!("Done.");
```

Comment:
One advantage of computing the hot rank in Rust is that it would be much easier to edit. Then we could even add a plugin system so different instances can use their own hot algorithm. But thats for the future.

Reply:
You can update multiple rows in a single query.

https://stackoverflow.com/a/18799497